src/nvim/msgpack_rpc/helpers.c

```diff
@@ -503,7 +510,6 @@ Object msgpack_rpc_handle_invalid_arguments(uint64_t channel_id,
                                             Array args,
                                             Error *error)
 {
```

Comment:
Yes it can be removed; but i'm thinking of avoiding the handlers by simply sending the error to the channel, this avoids having to add an error field to the event that conflicts in its handling with the handler function, also no "noop" handlers would be required and a request validation failure is handled in the same way already. What do you think?

Reply:
I'd imagine something like this
``` // check method arguments                                                                                   
  Array args = ARRAY_DICT_INIT;                                                                               
  if (!msgpack_rpc_to_array(msgpack_rpc_args(request), &args) && !ERROR_SET(&error)) {                        
    api_set_error(&error, kErrorTypeException, "Invalid method arguments");                                   
  }                                                                                                           
                                                                                                              
  // check method                                                                                             
  if (handler.fn == msgpack_rpc_handle_missing_method) {                                                      
    String m = method->via.bin.size > 0                                                                       
      ? cbuf_to_string(method->via.bin.ptr, method->via.bin.size)                                             
      : cstr_to_string("<empty>");                                                                            
    api_set_error(&error, kErrorTypeException, "Invalid method: %s", STRING_OBJ(m).data.string.data);         
  }                                                                                                           
                                                                                                              
  if (ERROR_SET(&error)) {                                                                                    
    channel_write(channel,                                                                                    
                  serialize_response(channel->id,                                                             
                                     request_id,                                                              
                                     &error,                                                                  
                                     NIL,                                                                     
                                     &out_buffer));                                                           
    api_clear_error(&error);                                                                                  
    return;                                                                                                   
  } 
```

but that fails on  
```
[----------] Running tests from test/functional/ex_cmds/ctrl_c_spec.lua
[ RUN      ] CTRL-C (mapped) interrupts :global
```
  can you imagine why that is so?

Reply:
The Test passes again after removing the build directory and building without ASAN.